cachi2/core/models/sbom.py

```diff
@@ -28,6 +28,25 @@ class ExternalReference(pydantic.BaseModel):
     type: Literal["distribution"] = "distribution"
 
 
+class PatchDiff(pydantic.BaseModel):
+    """A Diff inside a Patch."""
+
+    url: str
+
+
+class Patch(pydantic.BaseModel):
+    """A Patch inside a SBOM Component Pedigree."""
+
+    type: Literal["backport", "cherry-pick", "monkey", "unofficial"] = "unofficial"
```

Comment:
How can we determine which of the types: `unofficial`,`cherry-pick`, `backport` is the right one in order to improve or rather retain and not hinder the SBOM's accuracy?
I'm not questioning the use of Pedigree to record patch information, I'm just wondering whether by always reporting "unofficial" we're not going to cause more troubles for ourselves since I don't see a way for us to be able to detect it, it's a shame that the schema requires the type.
Can the choice of unofficial be recorded in an ADR at least? Because I feel like this is something worth recording from architectural POV.

Reply:
As I commented in the other [thread](https://github.com/containerbuildsystem/cachi2/pull/784#discussion_r1920169529), I think we can do a best-effort to assume that all patches added by the user (that is, that are present in the `package.json` file) are unnoficial. I think there'll be rare cases in which the maintainer of the library releases a patch to be used, if any.

We could see Yarn patches as unnoficial (since they are not written by the maintainers of the library being patched), but I'd like to differentiate them from user-added patches, since they are automatically added by Yarn, and are "official" if considering Yarn as the owner of the patch. Or, let's just omit them altogether, although I think the info could be somewhat useful (they do show up in `yarn.lock`).

Reply:
I don't think we need to go as formal as an ADR, let's just record the decision in a comment and the commit message, should be enough.

Reply:
What is the purpose of ADRs then? There clearly isn't one standard way on how one goes about recording patches in an SBOM, especially since there's a mandatory field we're not quite sure how to properly fill in based on the information we have.

Reply:
An ADR feels like overkill for a relatively minor decision like this, but :man_shrugging: 

There are two threads with discussion about the patch type, so I'm not sure which to use. My opinion is here: https://github.com/containerbuildsystem/cachi2/pull/784#discussion_r1922571758

Reply:
My final opinion is in the same thread. Let's make it simply "unofficial".

Reply:
> An ADR feels like overkill for a relatively minor decision like this, but 🤷‍♂️

Fair enough, although this isn't an internal implementation detail because it has an undocumented visible impact on users in form of our SBOM. Please just make sure that it's described accordingly at least in the commit msg, ideally with a tiny commentary in the code accompanying the change.